### PR TITLE
chore: migrate few APIs to brige API model

### LINKED ISSUE #31943: [Integrate PostgresDB] Task: Migrate `CustomActionCollectionRepositoryCEImpl.java` to use Bridge API - Part 1 / 3
[ref](https://theappsmith.slack.com/archives/CPQNLFHTN/p1710836011164579)

- [ ] Flux<ActionCollection> findByApplicationId(String applicationId, AclPermission aclPermission, Sort sort)
- [ ] Flux<ActionCollection> findByApplicationId(String applicationId, Optional<AclPermission> aclPermission, Optional<Sort> sort)
- [ ] Flux<ActionCollection> findByApplicationIdAndViewMode(String applicationId, boolean viewMode, AclPermission aclPermission)
- [ ] Flux<ActionCollection> findAllActionCollectionsByNameDefaultPageIdsViewModeAndBranch
- [ ] Flux<ActionCollection> findByPageId(String pageId, AclPermission aclPermission)

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
[**why ?** ](https://theappsmith.slack.com/archives/CPQNLFHTN/p1710836011164579)
**what ?** 
- Migrate first five APIs in `CustomActionCollectionRepositoryCEImp.java` to the Bridge API model . 
- Rest of the APIs (10 remaining) will be migrated in subsequent PRs. I am deliberately trying to push small, incremental and frequent changes for review .

fixes #31943 

## Automation

/ok-to-test tags="@tag.JS"

### :mag: Cypress test results

> [!IMPORTANT]  
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8387556951>
> Commit: `7d8c5a8348649283929651afe57c5e5efd3a3055`
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8387556951&attempt=1" target="_blank">Click here!</a>
> All cypress tests have passed 🎉🎉🎉

## Summary by CodeRabbit

- **Refactor**
    - Enhanced code readability and maintainability in server-side query handling by introducing a `BridgeQuery` abstraction.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java [hunk 1]
```diff
@@ -177,11 +162,11 @@ public Flux<ActionCollection> findByPageId(String pageId, AclPermission aclPermi
         String unpublishedPage = ActionCollection.Fields.unpublishedCollection_pageId;
         String publishedPage = ActionCollection.Fields.publishedCollection_pageId;
 
-        Criteria pageCriteria = new Criteria()
-                .orOperator(
-                        where(unpublishedPage).is(pageId), where(publishedPage).is(pageId));
+        BridgeQuery<ActionCollection> bridgeQuery = Bridge.or(
+                Bridge.<ActionCollection>equal(unpublishedPage, pageId),
+                Bridge.<ActionCollection>equal(publishedPage, pageId));
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java:167]: Same as above. The type can be removed. 
``
```suggestion
                Bridge.equal(publishedPage, pageId));
```

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java [hunk 2]
```diff
@@ -51,69 +52,58 @@ public Flux<ActionCollection> findByApplicationId(String applicationId, AclPermi
...
+            getBridgeQueryForFindAllActionCollectionsByNameDefaultPageIdsViewModeAndBranch(
+                    String branchName, boolean viewMode, String name, List<String> pageIds) {
         /**
          * TODO : This function is called by get(params) to get all actions by params and hence
          * only covers criteria of few fields like page id, name, etc. Make this generic to cover
          * all possible fields
          */
-        List<Criteria> criteriaList = new ArrayList<>();
-
+        BridgeQuery<ActionCollection> bridgeQuery = Bridge.<ActionCollection>query();
```
Comment (AnaghHegde) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java:97]: The type can be removed in the right hand side. It is redundant in my opinion.

**Review by AnaghHegde — COMMENTED** [review]
Minor comments, rest looks good to me.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ce/CustomActionCollectionRepositoryCEImpl.java]: Done.

**Review by AnaghHegde — APPROVED** [review]

**Pull request merged**